ner_v2/detectors/numeral/number/number_detection.py

```diff
@@ -144,9 +125,18 @@ def detect_entity(self, text, **kwargs):
         self.processed_text = self.text
         self.tagged_text = self.text
         number_data = self.language_number_detector.detect_number(self.processed_text)
-        self.number = number_data[0]
-        self.original_number_text = number_data[1]
-        return number_data
+        validated_number, validated_number_text = [], []
+        for number_value_dict, original_text in zip(number_data[0], number_data[1]):
+            if self.min_digit <= len(number_value_dict[NUMBER_DETECT_VALUE]) <= self.max_digit:
```

Comment:
Since `number_value_dict[NUMBER_DETECT_VALUE]` is `str`, a possible edge case would be leading zeros

min_digit = 1
max_digit = 2
`007` would fail

Reply:
```
In [3]: n.detect_entity('007')
Out[3]: ([{'unit': None, 'value': '7'}], ['007'])
```
As i am checking the case to convert any float of type 12.0 to int, This case also get handled